framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4813,30 +4964,62 @@ VkResult VulkanReplayConsumerBase::OverrideCreateSwapchainKHR(
         VkDevice                     device          = device_info->handle;
         const encode::DeviceTable*   device_table    = GetDeviceTable(device);
 
+        auto allocator = device_info->allocator.get();
+        assert(allocator != nullptr);
+
+        compatibility::SwapchainCreationInfo swapchain_creation_info{
+            device_info->handle,
+            physical_device,
+            replay_create_info,
+            { allocator,
+              CreateImageResource,
+              DestroyImageResource,
+              AllocateDeviceMemoryResource,
+              FreeDeviceMemoryResource,
+              BindImageResourceToDeviceMemoryResource },
```

Comment:
Decode is upper level library. Compatibility is lower level library. I don't know if it's common that lower level library use the function of upper level library?